apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-contentlets/dot-palette-contentlets.component.ts

```diff
@@ -0,0 +1,161 @@
+        private dotContentletEditorService: DotContentletEditorService
+    ) {}
+
+    ngOnInit(): void {
+        this._ESParams = {
+            itemsPerPage: 15,
+            lang: this.languageId || '1',
+            query: ''
+        };
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentTypeVariable?.currentValue) {
+            this.isFormContentType = changes?.contentTypeVariable?.currentValue === 'forms';
+
+            if (this.isFormContentType) {
+                this.paginationService.url = `v1/contenttype`;
+                this.paginationService.paginationPerPage = 15;
+                this.paginationService.sortField = 'modDate';
+                this.paginationService.setExtraParams('type', 'Form');
+                this.paginationService.sortOrder = 1;
+            }
+
+            this.loadData();
+        }
+    }
+
+    /**
+     * Loads data through pagination service
+     *
+     * @param LazyLoadEvent [event]
+     * @memberof DotPaletteContentletsComponent
+     */
+    loadData(event?: LazyLoadEvent): void {
+        if (this.isFormContentType) {
+            this.paginationService.setExtraParams(
+                'filter',
+                this.filterInput.searchInput.nativeElement.value
+            );
+            this.paginationService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    data.forEach((item) => (item.contentType = item.variable = 'FORM'));
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        } else {
+            this.paginatorESService
+                .get({
+                    ...this._ESParams,
+                    lang: this.languageId || '1',
+                    filter: this._filter || '',
+                    offset: (event && event.first) || 0,
+                    query: `+contentType: ${this.contentTypeVariable}`
+                })
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        }
+    }
+
+    /**
+     * Calls loads data function with a specific page to load
+     *
+     * @param LazyLoadEvent event
+     * @memberof DotPaletteContentletsComponent
+     */
+    paginate(event: LazyLoadEvent): void {
+        this.loadData(event);
+    }
+
+    /**
+     * Emits notification to show content type's component and clears
+     * component's local variables
+     *
+     * @param LazyLoadEvent event
+     * @memberof DotPaletteContentletsComponent
+     */
+    showContentTypesList(): void {
+        this.items = null;
+        this._filter = '';
+        this.filterInput.searchInput.nativeElement.value = '';
```

Comment:
Can we use `_filter` as model of the input and that way we don't have to touch the dom directly here?